microcosm/annotations.py

```diff
@@ -0,0 +1,106 @@
+"""
+Configuration annotations.
+
+"""
+from distutils.util import strtobool
+
+
+def boolean(value):
+    """
+    Annotate a value as boolean.
+
+    """
+    if value == "":
+        return False
+    return strtobool(value)
+
+
+def required(value):
+    """
+    Annotate a value as required.
+
+    """
+    if value is None:
+        raise Exception("Missing required value")
+    return value
+
+
+def must_override(metadata, value):
+    """
+    Annotate a value as required outside of unit testing.
+
+    """
+    if metadata.testing:
+        return value
+
+    return required(value)
+
+
+def evaluate(metadata, annotations, config):
+    """
+    Evaluate a set of annotations against a `Configuration`.
+
+    """
+    keys = set(annotations.keys()) | set(config.keys())
```

Comment:
how do we handle defined annotation for a missing config key?

Reply:
We'll fail right now.